### ISSUE TITLE
sort-json.ascendingSort in runOnSave seems to cause an infinite autoSave loop

### ISSUE THREAD

**Issue by mlao-pdx** [R35007/sort-json-vscode-extension#11]
Hi,

Thank you for creating the sort-json extension.

I _think_ that sort-json overwrites the file content, regardless of whether the sort has resulted in changes to the file. This causes the autoSave to trigger, which in turn triggers my RunOnSave ("pucelle.run-on-save") config, which runs sort-json.ascendingSort, which cause autoSave to trigger, which...

I have other formatters and linters turned off so that they cannot interfere.

Assuming that my suspicion is valid, and given that two or more identically configured, consecutive sorts _must_ result in an unchanging file; would it be possible to have sort-json check if its output has changed the input, and if so, only then write it back to the editor?

A `.vscode/settings.json` configuration that should reproduce the scenario (NOTE: .vscode is skipped by globMatch, so it will not show on settings.json itself):

```jsonc
...
  "runOnSave.commands": [
    {
      "command": "sort-json.ascendingSort",
      "globMatch": "**/*.json",
      "runIn": "vscode"
    }
  ],
...
  "sort-json.settings.contextMenu": {
    "ascendingSort": true,
    "sortJsonSubMenu": false
  },
  "sort-json.settings.isCaseSensitive": true,
  "sort-json.settings.sortMode": "Objects Only",
...
```

**Comment by R35007** [comment]
@mlao-pdx,

Thanks for pointing out this issue. 
I have fixed and released a fix version 17.0.2. 
Please update the extension and try it again.

**Comment by mlao-pdx** [comment]
> @mlao-pdx,
> 
> Thanks for pointing out this issue. I have fixed and released a fix version 17.0.2. Please update the extension and try it again.

Tested 17.0.2. Works like a charm and notifies "Already sorted".
